### PR TITLE
refactor: add new line to the end of  all kotlin files

### PR DESCRIPTION
**Pull request by mayokunthefirst**
Fixes issue #235

### CODE REVIEW

**Comment by arkivanov** [comment]
Build failed:
Task :utils:detekt
......................................
38 kotlin files were analyzed.
Ruleset: style - 5min debt
	NewLineAtEndOfFile - [LockExt.kt] at /home/travis/build/badoo/Reaktive/utils/src/commonMain/kotlin/com/badoo/reaktive/utils/lock/LockExt.kt:18:4

**Comment by mayokunthefirst** [comment]
Hello @arkivanov, I would love to ask if you know why the build keeps failing. I have also tried building at my end, but got the same build failure. Thank you

**Comment by arkivanov** [comment]
@mayokunthefirst It fails because there is no empty new line at the end of the "LockExt.kt" file

File: utils/src/commonMain/kotlin/com/badoo/reaktive/utils/lock/LockExt.kt [hunk 1]
```diff
@@ -15,4 +15,4 @@ inline fun <T> Lock.use(block: (Lock) -> T): T =
     } finally {
         destroy()
     }
-    
\ No newline at end of file
+
```
Comment (CherryPerry) [inline utils/src/commonMain/kotlin/com/badoo/reaktive/utils/lock/LockExt.kt]: Detekt failed because of this spaces. Keep one empty line without spaces.

Comment (CherryPerry) [reply utils/src/commonMain/kotlin/com/badoo/reaktive/utils/lock/LockExt.kt]: Also you can check it locally before push by invoking `detekt` Gradle task.

**Comment by CherryPerry** [comment]
@arkivanov Do squash merge instead of regular merge please.

**Comment by CherryPerry** [comment]
@mayokunthefirst And small question. IDEA/Android Studio has alert about different line separators in committed files, did you receive such warning, when committed changes?

**Comment by mayokunthefirst** [comment]
> @mayokunthefirst And small question. IDEA/Android Studio has alert about different line separators in committed files, did you receive such warning, when committed changes?

Hello @CherryPerry I used VSCode as opposed to IDEA/Android Studio because of it's well structured file hierarchy system and built the project on Travis-CI. Seeing VSCode cannot throw such warnings, I have switched to IDEA/Android Studio.

File: utils/src/commonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt [hunk 2]
```diff
@@ -3,5 +3,8 @@ package com.badoo.reaktive.utils
 expect fun <T> T.freeze(): T
 
 expect fun <T: Any> T.ensureNeverFrozen(): T
+<<<<<<< HEAD
```
Comment (arkivanov) [inline utils/src/commonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: Could you please resolve all the conflicts?

File: utils/src/jvmJsCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt [hunk 3]
```diff
@@ -3,5 +3,8 @@ package com.badoo.reaktive.utils
 actual fun <T> T.freeze(): T = this
 
 actual fun <T : Any> T.ensureNeverFrozen(): T = this
+<<<<<<< HEAD
```
Comment (arkivanov) [inline utils/src/jvmJsCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: And here

File: utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt [hunk 4]
```diff
@@ -11,5 +11,8 @@ actual fun <T: Any> T.ensureNeverFrozen(): T {
 
     return this
 }
+<<<<<<< HEAD
```
Comment (arkivanov) [inline utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: Conflict

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/ToList.kt [hunk 5]
```diff
@@ -4,6 +4,9 @@ import com.badoo.reaktive.single.Single
 import com.badoo.reaktive.utils.isFrozen
 
 fun <T> Observable<T>.toList(): Single<List<T>> =
+<<<<<<< HEAD
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/ToList.kt]: And here

File: utils/src/commonMain/kotlin/com/badoo/reaktive/utils/lock/LockExt.kt (hunk 1 above)
Comment (mayokunthefirst) [reply utils/src/commonMain/kotlin/com/badoo/reaktive/utils/lock/LockExt.kt]: Thank you @CherryPerry , I have done this 👍

File: utils/src/commonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt (hunk 2 above)
Comment (mayokunthefirst) [reply utils/src/commonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: Yes, I have 👍

File: utils/src/jvmJsCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt (hunk 3 above)
Comment (mayokunthefirst) [reply utils/src/jvmJsCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: Done 👍

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/ToList.kt (hunk 5 above)
Comment (mayokunthefirst) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/ToList.kt]: Done 👍

File: utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt (hunk 4 above)
Comment (mayokunthefirst) [reply utils/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Freeze.kt]: Resolved 👍

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/CompositeDisposable.kt [hunk 6]
```diff
@@ -1,56 +1,32 @@
...
-
 /**
  * Thread-safe collection of [Disposable]
  */
-class CompositeDisposable : Disposable {
-
-    private val list = AtomicReference<List<Disposable>?>(emptyList())
-    override val isDisposed: Boolean get() = list.value == null
+@Suppress("EmptyDefaultConstructor")
+expect class CompositeDisposable() : Disposable {
```
Comment (CherryPerry) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/CompositeDisposable.kt]: Why these changes in this diff?

Comment (mayokunthefirst) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/CompositeDisposable.kt]: Hello @CherryPerry , there was an error when I did a `git merge` trying to pull in new commits from this original repo into my fork. I had to fix the merge errors. Currently, there is no difference as I fixed all errors.

**Comment by arkivanov** [comment]
Hello @mayokunthefirst, unfortunately we can't accept this PR because it contains unrelated commits (e.g. my commits from another branches). I see you are working directly on master, but If you want to pull-request, then I would suggest you to work on a branch. If you still would like to help us with this task, then please do something like this:

- save your current work to a temporary branch: `git checkout -b temp`
- add upstream remote: `git remote add upstream https://github.com/badoo/Reaktive.git`
- fetch the upstream: `git fetch upstream`
- checkout upstream's master branch: `git checkout upstream/master`
- checkout a new branch for PR: `git checkout -b add-eof-lines`
- cherry-pick all your commits from `temp` branch, from oldest to newest: `git cherry-pick <commit_hash>` (for every cherry-pick if there are conflicts: resolve all conflicts, `git add --all`, `git cherry-pick --continue`)
- create a new PR from `add-eof-lines` to our `master`

Closing this PR.

**Comment by mayokunthefirst** [comment]
> Hello @mayokunthefirst, unfortunately we can't accept this PR because it contains unrelated commits (e.g. my commits from another branches). I see you are working directly on master, but If you want to pull-request, then I would suggest you to work on a branch. If you still would like to help us with this task, then please do something like this:
> 
> * save your current work to a temporary branch: `git checkout -b temp`
> * add upstream remote: `git remote add upstream https://github.com/badoo/Reaktive.git`
> * fetch the upstream: `git fetch upstream`
> * checkout upstream's master branch: `git checkout upstream/master`
> * checkout a new branch for PR: `git checkout -b add-eof-lines`
> * cherry-pick all your commits from `temp` branch, from oldest to newest: `git cherry-pick <commit_hash>` (for every cherry-pick if there are conflicts: resolve all conflicts, `git add --all`, `git cherry-pick --continue`)
> * create a new PR from `add-eof-lines` to our `master`
> 
> Closing this PR.

Alright then, i'll still love to work on the task. I'll get on it right away 👍
